manager.go

```diff
@@ -42,4 +42,14 @@ type Manager interface {
 	// a set that exactly matches the request, or a superset of it. If an error occurs, it returns nil and
 	// the error.
 	FindRequestCandidates(r *Request) (Policies, error)
+
+	// FindPoliciesForSubject returns policies that could match the resource. It either returns
+	// a set that exactly matches the request, or a superset of it. If an error occurs, it returns nil and
+	// the error.
+	FindPoliciesForSubject(subject string) (Policies, error)
```

Comment:
I know that this is kind of strange, because we already have `GetPolicy` and stuff like that, but it has been a while since this interface was defined. Best practice in Go is to define interfaces at the consumer rather than the producer. These methods are really only used in the tests at the moment (to hydrate the stores).

I'm not generally against exposing these methods as an interface here - but maybe it would make sense to have two interfaces `RequestCandidateFinder` and `Manager`? The former would only expose `FindRequestCandidates` (needed by `Ladon`) and the latter would expose management methods including these ones here? This would obviously be a breaking change but it would clean this up a bit.

What do you think?

Reply:
I have looked a bit to other code to try to understand the implications.

I agree completely that managing policies and actually authorizing could very well be two separate interfaces.
I struggle to visualize how that would be done nicely though, because I feel that most (all?) implementations should prefer keeping the code for management and for authentication close to each other since it's touching the same data. So, the postgres `Manager` and `RequestCandidateFinder` should probably be the same struct implementing both interfaces IMO.

Maybe a clean way to get this would be to introduce the new interfaces, but change SQLManager to something like:

    type SQLPolicyDB struct {
	db       *sqlx.DB
	database string
    }
    
    // ... All the same methods that are in SQLManager now, give or take

    func (s *SQLPolicyDB) GetManager() Manager {
      return s
    }

    func (s *SQLPolicyDB) GetRequestCandidateFinder() RequestCandidateFinder {
      return s
    } 

Then the change to dependent systems maybe is not that bad?